### PR TITLE
Refactoring fractal dimension

### PR DESCRIPTION
**Pull request by martinjanssens**
I started off with an easy one. @leifdenby I'll try to stay as close to the structure you've been using as I can, and will give a shout when it might be good to have a look.

### CODE REVIEW

**Comment by martinjanssens** [comment]
I think this is good to go content-wise. I'm not sure I've adhered to all the new standards, but have done my best. Curious to hear what you think, Leif!

File: requirements.txt [hunk 1]
```diff
@@ -12,3 +12,5 @@ scikit-learn>=0.22.1
 matplotlib>=3.1.3
 pyhdf>=0.10.1
 seaborn>=0.10.0
+
+hilbertcurve>=2.0.5
```
Comment (leifdenby) [inline requirements.txt:16]: We could put this in a `requirements-test.txt` file that way people who aren't running the tests won't get this installed on their system? There is a `setup.py` option called `tests_require`, but that appears to be deprecated now and I don't know what the best option is...

File: tests/test_fractal_dimension.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import numpy as np
+import pytest
+from hilbertcurve.hilbertcurve import HilbertCurve
+
+import cloudmetrics
+
+
+def test_hilbert_curve():
```
Comment (leifdenby) [inline tests/test_fractal_dimension.py:8]: So awesome you added this!

**Review by leifdenby — COMMENTED** [review]
The only change I'd make it put the `hilbertcurve` package into its own `requirements-test.txt` file (and make sure that gets added to the github action).

Beautiful work!

File: requirements.txt (hunk 1 above)
Comment (martinjanssens) [reply requirements.txt]: That's really smart. I've added this.

File: tests/test_fractal_dimension.py (hunk 2 above)
Comment (martinjanssens) [reply tests/test_fractal_dimension.py:8]: It's nice that someone actually went through the trouble of writing a Hilbert curve package, it's super easy to use :)

**Pull request merged**